capstone/src/main/python/main.py

```diff
@@ -0,0 +1,123 @@
+from flask import Flask
+from google.cloud import datastore
+from sklearn.tree import DecisionTreeRegressor
+import numpy as np
+import pandas as pd
+app = Flask(__name__)
+
+
+@app.route('/regression-analysis')
+def main():
+    df = load_dataframe()
+    fit_regressor = train_model(df)
+    add_predictions(fit_regressor)
+
+
+def load_dataframe():
+    datastore_client = datastore.Client()
+    query = datastore_client.query(kind='Response')
+    results = list(query.fetch())
+
+    # Create Data Frame from entities
+    direct_experiences = []  # Experience with police (yes or no)
+    genders = []
+    ages = []
+    scores = []
+
+    for response in results:
+        direct_experiences.append(response['directExperience'])
+        genders.append(response['gender'])
+        ages.append(response['ageRange'])
+        scores.append(response['score'])
+
+    data_points = {'directExperience': direct_experiences, 'gender': genders,
+                   'ageRange': ages, 'score': scores}
+
+    df = pd.DataFrame(data_points, 
+                      columns=['directExperience', 'gender', 'ageRange', 'score'])
+
+    return df
+ 
+
+def train_model(df):
+    x = df[['directExperience', 'gender', 'ageRange']]
+    y = df['score']
+
+    y = y.values.reshape(-1, 1)
+
+    # Encode categorical variables into numbers
+    x_dum = pd.get_dummies(x, prefix_sep='_')
+
+    x_matrix = x_dum.to_numpy()
+
+    regressor = DecisionTreeRegressor()
+    regressor.fit(x_matrix, y)
+
+    return regressor
+
+
+# This function takes values
+# for each feature and returns an
+# encoded array that represents those ordered values
+# in a binary form (needed for regressor predict())
+def getEncodedArray(direct_experience, gender, ageRange):
```

Comment:
snake case? get_encoded_array

Reply:
I'll add that change to ensure consistency